src/services/item/plugins/invitation/repository.ts

```diff
@@ -1,73 +1,107 @@
+export class InvitationRepository extends MutableRepository<Invitation, UpdateInvitationBody> {
+  constructor(manager?: EntityManager) {
+    super(DEFAULT_PRIMARY_KEY, Invitation, manager);
+  }
+
+  async addOne({ partialInvitations, itemPath, creatorId }: CreateInvitationBody) {
+    this.throwsIfParamIsInvalid('itemPath', itemPath);
+    this.throwsIfParamIsInvalid('creatorId', creatorId);
+
+    return await super.insert({
+      ...partialInvitations,
+      item: { path: itemPath },
+      creator: { id: creatorId },
+    });
+  }
+
+  async getOne(id: string) {
+    this.throwsIfPKIsInvalid(id);
+    return await this.repository
+      .createQueryBuilder('invitation')
+      .innerJoinAndSelect('invitation.item', 'item')
+      .where('invitation.id = :id', { id })
+      .getOne();
+  }
+
   /**
    * Get invitation by id or null if it is not found
    * @param id Invitation id
    */
-  async get(id: string, actor?: Actor): Promise<Invitation> {
-    const query = this.createQueryBuilder('invitation')
-      .innerJoinAndSelect('invitation.item', 'item')
-      .where('invitation.id = :id', { id });
+  async getOneByIdAndByCreatorOrThrow(id: string, creatorId: CreatorId): Promise<Invitation> {
+    this.throwsIfPKIsInvalid(id);
+    this.throwsIfParamIsInvalid('creatorId', creatorId);
 
-    if (actor) {
-      query.innerJoinAndSelect('invitation.creator', 'creator');
-    }
-    const invitation = await query.getOne();
-    if (!invitation) {
-      throw new InvitationNotFound(id);
-    }
-    return invitation;
-  },
-
-  /**
-   * Get invitations map by id
-   * @param ids Invitation ids
-   */
-  async getMany(ids: string[], actor?: Member): Promise<ResultOf<Invitation>> {
-    const query = this.createQueryBuilder('invitation')
+    const entity = await this.repository
+      .createQueryBuilder('invitation')
       .innerJoinAndSelect('invitation.item', 'item')
-      .where('invitation.id IN (:...ids)', { ids });
+      .innerJoinAndSelect('invitation.creator', 'creator')
+      .where('invitation.id = :id', { id })
+      .andWhere('invitation.creator_id = :creatorId', { creatorId })
+      .getOne();
 
-    if (actor) {
-      query.innerJoinAndSelect('invitation.creator', 'creator');
+    if (!entity) {
+      throw new InvitationNotFound(id);
     }
-    const invitations = await query.getMany();
 
-    return mapById({
-      keys: ids,
-      findElement: (invId) => invitations.find(({ id }) => id === invId),
-      buildError: (id) => new InvitationNotFound(id),
-    });
-  },
+    return entity;
+  }
 
   /**
    * Get invitations for item path and below
    * @param itemPath Item path
    */
-  async getForItem(itemPath: string): Promise<Invitation[]> {
-    return this.createQueryBuilder('invitation')
-      .innerJoinAndSelect('invitation.item', 'item')
-      .where(':path <@ item.path', { path: itemPath })
-      .getMany();
-  },
+  async getManyByItem(itemPath: ItemPath) {
```

Comment:
Because we are using `getOne...` to specify the unique returned row, what do you think about using `get...` for the queries that return multiple rows?

```suggestion
  async getByItem(itemPath: ItemPath) {
```